api/client.go

```diff
@@ -52,6 +62,13 @@ func (cfg *Config) roundTripper() http.RoundTripper {
 	return cfg.RoundTripper
 }
 
+func (cfg *Config) client() http.Client {
```

Comment:
This method needs to take account of the round tripper. We should consume the given roundtripper through config, no? Consider calling `cfg.roundTripper()` accordingly.

Reply:
Hey, sorry there seems to be a confusion. The first solution was to use the RoundTripper provided in the configuration and you suggested to follow same pattern as `roundTripper()`
https://github.com/prometheus/client_golang/pull/1025#discussion_r847285988

Reply:
Sorry, for creating the confusion. I guess I've missed the fact when I wrote it :see_no_evil: We should use when the roundTripper specified.